Source/ARTRealtimePresence.m

```diff
@@ -15,26 +15,65 @@
 #import "ARTStatus.h"
 #import "ARTPresence+Private.h"
 
+#pragma mark - ARTRealtimePresenceQuery
+
+@implementation ARTRealtimePresenceQuery
+
+- (instancetype)initWithLimit:(NSUInteger)limit clientId:(NSString *)clientId connectionId:(NSString *)connectionId {
+    self = [super initWithLimit:limit clientId:clientId connectionId:connectionId];
+    if (self) {
+        _waitForSync = true;
+    }
+    return self;
+}
+
+@end
+
+#pragma mark - ARTRealtimePresence
+
+@interface ARTRealtimePresence ()
+
+@property (readonly, getter=getChannel) ARTRealtimeChannel *channel;
+
+@end
+
 @implementation ARTRealtimePresence
 
 - (instancetype)initWithChannel:(ARTRealtimeChannel *)channel {
-    return [super initWithChannel:channel];
+    self = [super initWithChannel:channel];
+    return self;
 }
 
-- (ARTRealtimeChannel *)channel {
+- (ARTRealtimeChannel *)getChannel {
     return (ARTRealtimeChannel *)super.channel;
 }
 
-- (void)get:(ARTRealtimePresenceQuery *)query callback:(void (^)(ARTPaginatedResult<ARTPresenceMessage *> * _Nullable, NSError * _Nullable))callback {
-    [[self channel] throwOnDisconnectedOrFailed];
-    [super get:query callback:callback];
+- (void)get:(void (^)(NSArray<ARTPresenceMessage *> *, ARTErrorInfo *))callback {
+    [self get:[[ARTRealtimePresenceQuery alloc] init] callback:callback];
+}
+
+- (void)get:(ARTRealtimePresenceQuery *)query callback:(void (^)(NSArray<ARTPresenceMessage *> *, ARTErrorInfo *))callback {
+    [self.channel throwOnDisconnectedOrFailed];
+    [self.channel attach:^(ARTErrorInfo *error) {
+        if (query.waitForSync) {
```

Comment:
Oh, I see. So [ARTRestPresence.get](https://github.com/ably/ably-ios/blob/957b709f8e7cbb9ead8bfdd7d028ffd9e6826854/Source/ARTRestPresence.m#L31) is also missing those members, right? Maybe I will do it in another PR, WDYT?